src/Api/Request/Search/Search.php

```diff
@@ -0,0 +1,89 @@
+<?php declare(strict_types=1);
+
+namespace PeeHaa\AsyncTwitter\Api\Request\Search;
+
+use PeeHaa\AsyncTwitter\Api\Request\BaseRequest;
+
+/**
+ * @link https://dev.twitter.com/rest/reference/get/search/tweets
+ */
+class Search extends BaseRequest
+{
+    const METHOD = 'GET';
+    const ENDPOINT = '/search/tweets.json';
+
+    private $resultTypes = ['mixed', 'recent', 'popular'];
+
+    public function __construct(string $searchString)
+    {
+        parent::__construct(self::METHOD, self::ENDPOINT);
+
+        $this->parameters['q'] = $searchString;
+    }
+
+    //use string instead of float
+    public function setGeocode(string $latitude, string $longitude, string $radius): Search
```

Comment:
Probably cleaner to drop all the get* prefixes for the methods in this class because all methods are just setters.